### PR TITLE
LOC CHECKIN | dotnet/roslyn master-vs-deps | 20181123

### PR DESCRIPTION
**Pull request by v-zbsail**

### CODE REVIEW

File: src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf [hunk 1]
```diff
@@ -4,312 +4,312 @@
...
       <trans-unit id="ERR_AwaitForEachMissingMemberWrongAsync">
         <source>Asynchronous foreach statement cannot operate on variables of type '{0}' because '{0}' does not contain a public instance definition for '{1}'. Did you mean 'foreach' rather than 'await foreach'?</source>
-        <target state="new">Asynchronous foreach statement cannot operate on variables of type '{0}' because '{0}' does not contain a public instance definition for '{1}'. Did you mean 'foreach' rather than 'await foreach'?</target>
+        <target state="translated">Eine asynchrone foreach-Anweisung kann nicht für Variablen vom Typ "{0}" verwendet werden, weil "{0}" keine öffentliche Instanzendefinition für "{1}" enthält. Meinten Sie "foreach" statt "await foreach"?</target>
         <note />
       </trans-unit>
       <trans-unit id="ERR_BadDynamicAwaitForEach">
         <source>Cannot use a collection of dynamic type in an asynchronous foreach</source>
-        <target state="new">Cannot use a collection of dynamic type in an asynchronous foreach</target>
+        <target state="translated">In einer asynchronen foreach-Schleife kann keine Sammlung vom Typ "Dynamisch" verwendet werden.</target>
```
Comment (jhinder) [inline src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf]: "vom dynamischen Typ" or "vom Typ "dynamic"" would be better I think.

File: src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf [hunk 2]
```diff
@@ -1949,12 +1949,12 @@
       </trans-unit>
       <trans-unit id="ERR_IllegalStatement">
         <source>Only assignment, call, increment, decrement, await, and new object expressions can be used as a statement</source>
-        <target state="needs-review-translation">Nur assignment-, call-, increment-, decrement- und new object-Ausdrücke können als Anweisung verwendet werden.</target>
+        <target state="translated">Nur assignment-, call-, increment-, decrement- und new object-Ausdrücke können als Anweisung verwendet werden.</target>
```
Comment (jhinder) [inline src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf]: "await" is missing from the list.

File: src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf [hunk 3]
```diff
@@ -6415,7 +6415,7 @@ Unterstützen Sie den Compiler bei der Unterscheidung zwischen den Methoden. Daz
       </trans-unit>
       <trans-unit id="ERR_BadAsyncReturn">
         <source>The return type of an async method must be void, Task, Task&lt;T&gt;, a task-like type, or IAsyncEnumerable&lt;T&gt;</source>
-        <target state="needs-review-translation">Der Rückgabetyp einer Async-Methode muss "void", "Task" oder "Task&lt;T&gt;" sein</target>
+        <target state="translated">Der Rückgabetyp einer Async-Methode muss "void", "Task" oder "Task&lt;T&gt;" sein</target>
```
Comment (jhinder) [inline src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf]: `a task-like type, or IAsyncEnumerable&lt;T&gt;` is untranslated.

**Comment by janbaltus** [comment]
We are working on fixing those issues

**Comment by janbaltus** [comment]
PR updated with fixed translation.

File: src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf [hunk 4]
```diff
@@ -5247,7 +5247,7 @@ Ein catch()-Block nach einem catch (System.Exception e)-Block kann nicht-CLS-Aus
       </trans-unit>
       <trans-unit id="ERR_BadAsyncReturnExpression">
         <source>Since this is an async method, the return expression must be of type '{0}' rather than 'Task&lt;{0}&gt;'</source>
-        <target state="translated">Da es sich um eine asynchrone Methode handelt, muss der Rückgabeausdruck vom Typ "{0}" anstelle von "Task&lt;{0}&gt;" sein.</target>
+        <target state="translated">Der Rückgabetyp einer async-Methode muss "void", "Task", "Task&lt;T&gt;", ein taskähnlicher Typ oder "IAsyncEnumerable&lt;T&gt;" sein.</target>
```
Comment (jhinder) [inline src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf]: This translation was fine, but [`ERR_BadAsyncReturn`](https://github.com/dotnet/roslyn/pull/31327/files#diff-f7c7796cace976d0368b7b23a2e7469dR6418) still has the [wrong translation](https://github.com/dotnet/roslyn/pull/31327/files#r236001755).

Comment (janbaltus) [reply src/Compilers/CSharp/Portable/xlf/CSharpResources.de.xlf]: Please check this issue again. Fixed and updated PR.

**Comment by tmeschter** [comment]
This PR has been superseded by https://github.com/dotnet/roslyn/pull/32616.

**Pull request closed without merging**